### ISSUE TITLE
[BUG] Crash in [FPRHash fingerprintForString:]

### ISSUE THREAD

**Issue by akshetpandey** [firebase/firebase-ios-sdk#495]
### [READ] Step 1: Are you in the right place?
Yes

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: **Version 9.1 (9B55)**
  * Firebase SDK version: **Firebase/Core (4.5.0)**
  * Firebase Component: **Perf** (?)
  * Component version: **Firebase/Performance (4.5.0)**

### [REQUIRED] Step 3: Describe the problem
Crash in instrumentation during auth flow (may be?)

I have encountered two crash reports related to this problem:

**Stack Trace 1:**
```
Exception name: EXC_BAD_ACCESS / KERN_INVALID_ADDRESS
1 libsystem_c.dylib strlen + 4524
2 thriller +[FPRHash fingerprintForString:] + 2075000
3 thriller +[FPRHash fingerprintForString:] + 2075000
4 thriller +[FPRFeatureControl featureEnabled:] + 2072636
5 thriller __InstrumentSessionWithConfigurationDelegateDelegateQueue_block_invoke + 2136944
6 thriller -[GTMSessionFetcher beginFetchMayDelay:mayAuthorize:] (GTMSessionFetcher.m:622)
7 thriller -[FIRAuthBackendRPCIssuerImplementation asyncPostToURLWithRequestConfiguration:URL:body:contentType:completionHandler:] + 1125332
8 thriller -[FIRAuthBackendRPCImplementation postWithRequest:response:callback:] + 1132184
9 thriller -[FIRAuthBackendRPCImplementation secureToken:callback:] + 1128576
10 thriller +[FIRAuthBackend secureToken:callback:] + 1123220
11 thriller -[FIRSecureTokenService requestAccessToken:] + 1087152
12 thriller __65-[FIRSecureTokenService fetchAccessTokenForcingRefresh:callback:]_block_invoke + 1085840
13 thriller __38-[FIRAuthSerialTaskQueue enqueueTask:]_block_invoke + 1077828
14 libdispatch.dylib _dispatch_call_block_and_release + 4228
15 libdispatch.dylib _dispatch_client_callout + 4164
16 libdispatch.dylib _dispatch_queue_serial_drain$VARIANT$mp + 44612
17 libdispatch.dylib _dispatch_queue_invoke$VARIANT$mp + 47060
18 libdispatch.dylib _dispatch_queue_serial_drain$VARIANT$mp + 44284
19 libdispatch.dylib _dispatch_queue_invoke$VARIANT$mp + 47060
20 libdispatch.dylib _dispatch_root_queue_drain_deferred_wlh$VARIANT$mp + 49660
21 libdispatch.dylib _dispatch_workloop_worker_thread$VARIANT$mp + 83100
22 libsystem_pthread.dylib _pthread_wqthread + 4044
```

**Stack Trace 2:**
```
Exception name: EXC_BAD_ACCESS / KERN_INVALID_ADDRESS
1 libsystem_c.dylib strlen + 4556
2 thriller +[FPRHash fingerprintForString:] + 2075000
3 thriller +[FPRHash fingerprintForString:] + 2075000
4 thriller +[FPRFeatureControl featureEnabled:] + 2072636
5 thriller __InstrumentSessionWithConfigurationDelegateDelegateQueue_block_invoke + 2136944
6 thriller -[FIRNetworkURLSession sessionIDFromAsyncPOSTRequest:completionHandler:] + 1193164
7 thriller -[FIRNetwork postURL:payload:queue:usingBackgroundSession:completionHandler:] + 1185312
8 thriller -[FIRClearcutLogger sendNextPendingRequestWithCompletionHandler:] + 1223004
9 thriller __65-[FIRClearcutLogger sendNextPendingRequestWithCompletionHandler:]_block_invoke + 1223628
10 libdispatch.dylib _dispatch_call_block_and_release + 4228
11 libdispatch.dylib _dispatch_client_callout + 4164
12 libdispatch.dylib _dispatch_root_queue_drain + 58228
13 libdispatch.dylib _dispatch_worker_thread3 + 57100
14 libsystem_pthread.dylib _pthread_wqthread + 4396
```

#### Steps to reproduce:
The only thing I know for sure is the crash happens long after the app has been backgrounded.
The last event log that I get for the session is a background event and the app crash a few hours after that.

#### Relevant Code:
Internal firebase crash, the relevant code is not open source.
I can't figure out which, if any, of my code is related to this.

**Comment by paulb777** [comment]
Tracked internally at b/69804394

**Comment by mikehaney24** [comment]
This will be fixed in the next version of the SDK, will close when that releases.

**Comment by mikehaney24** [comment]
This issue should be fixed in the latest FirebasePerformance pod, v1.1.0. Please feel free to reopen if that's not the case!
